tests/Task.test.ts

```diff
@@ -603,4 +631,41 @@ describe('toggle done', () => {
             todaySpy.mockClear();
         },
     );
+
+    it('supports recurrence rule after a due date', () => {
+        // Arrange
+        const line =
+            '- [ ] this is a task 🗓 2021-09-12 ✅ 2021-06-20 🔁 every day';
+        const path = 'this/is a path/to a/file.md';
+        const sectionStart = 1337;
+        const sectionIndex = 1209;
+        const precedingHeader = 'Eloquent Section';
+
+        // Act
+        const task = Task.fromLine({
+            line,
+            path,
+            sectionStart,
+            sectionIndex,
+            precedingHeader,
+        });
+
+        // Assert
+        expect(task).not.toBeNull();
+        expect(task!.dueDate).not.toBeNull();
+        expect(
+            task!.dueDate!.isSame(moment('2021-09-12', 'YYYY-MM-DD')),
+        ).toStrictEqual(true);
+
+        const nextTask: Task = task!.toggle()[0];
+        expect({
+            nextDue: nextTask.dueDate?.format('YYYY-MM-DD'),
+            nextScheduled: nextTask.scheduledDate?.format('YYYY-MM-DD'),
+            nextStart: nextTask.startDate?.format('YYYY-MM-DD'),
+        }).toMatchObject({
+            nextDue: '2021-09-13',
+            nextScheduled: undefined,
+            nextStart: undefined,
+        });
+    });
```

Comment:
In general, I feel that the test coverage is a bit weak for such a fundamental change in behaviour.

It tests 2 new happy paths, and doesn't really try and break the behaviour, or push it to any kind of limits.

One way to improve the coverage of the new behaviour would be to modify some of the existing tests by mutating the order of some of the fields, to show that the behaviour is unchanged even if users add things in a different order...

Reply:
Done: added more test cases and modified several existing tests for the new behavior.